File: py/handshake/validator.py
from jsonschema import validate, ValidationError
from .schema import HANDSHAKE_SCHEMA

def validate_handshake(envelope: dict):
    try:
        validate(instance=envelope, schema=HANDSHAKE_SCHEMA)
    except ValidationError as e:
        raise ValueError(f"Invalid HandshakeEnvelope: {e.message}")

File: ts/handshake/validator.ts
import Ajv from "ajv";
import schema from "./handshake.schema.json";

const ajv = new Ajv({ allErrors: true });
const validate = ajv.compile(schema);

export function validateHandshakeEnvelope(envelope: unknown) {
  const valid = validate(envelope);
  if (!valid) {
    throw new Error(
      "Invalid HandshakeEnvelope: " +
      JSON.stringify(validate.errors, null, 2)
    );
  }
  return true;
}